packages/flutter/lib/src/cupertino/nav_bar.dart

```diff
@@ -1125,44 +1164,104 @@ class _CupertinoSliverNavigationBarState extends State<CupertinoSliverNavigation
 
   @override
   Widget build(BuildContext context) {
+    if (widget.bottom is _NavigationBarSearchField && !atTop) {
+      effectiveBottom = GestureDetector(
+        behavior: HitTestBehavior.translucent,
+        child: widget.bottom,
+        onTap: () {
+          setState(() {
+            atTop = true;
```

Comment:
The entire logic on where to set `atTop` feels hacky to me, are there any previous examples or tips I could draw from? The major purpose of `atTop` is just to track if the search bar is in its default location or if it is at the top because it has been tapped/focused.

Reply:
Listening to the AnimationStatus (addStatusListener) might do the trick. That way you can listen for when the animation starts and ends, and change the atTop then. There are likely a bunch of examples of this, Snackbar & ScaffoldMessenger come to mind, check _handleSnackBarStatusChanged to see the animation status being followed to trigger changes after it completes. :)